crash-handler/src/main/java/org/odk/collect/crash_handler/CrashHandler.kt

```diff
@@ -0,0 +1,88 @@
+package org.odk.collect.crash_handler
+
+import android.content.Context
+import org.odk.collect.androidshared.data.getState
+import kotlin.system.exitProcess
+
+class CrashHandler(private val processKiller: Runnable = Runnable { exitProcess(0) }) {
+
+    private var conditionFailure: String? = null
+
+    fun registerCrash(context: Context, crash: Throwable) {
+        getPreferences(context).edit().putString(KEY_CRASH, crash.message ?: "").apply()
```

Comment:
Shouldn't we keep this in meta prefs?

Reply:
Yeah I'd initially though about it working that way (or creating a new "temp" settings). The problem is that we can't pass/inject a `SettingsProvider` from `collect_app` the Dagger graph (the `AppDependencyComponent`) might not have been created when crashes occur or are displayed. Given that, I decided to have `CrashHandler` just use raw Android APIs and give it its own `SharedPreferences` bucket. Does that make sense?

Reply:
Yes but maybe we should add a comment I'm sure I will forget the reason soon and will be thinking about the same next time I came across this code.